### DISCUSSION TITLE
Create an adapter for Spring Web

### DISCUSSION THREAD

**Discussion by Thihup** [piranhacloud/piranha#1983] [Ideas]
The main server for Spring framework is Tomcat. They also have the possibility to use Jetty instead of Tomcat.

Now that Spring 6 will use Jakarta EE 9 and Java 17, maybe we could think of providing a way to use Piranha as an embedded server for Spring.

There are some things to consider:
Would it be in this repository or should this be in the official repository. Maybe it would be harder to get there first, so meanwhile we could use this repository or create another, only to start experimenting and maturing this idea.

While this might sound crazy, I guess it would be a good way to test our implementation besides our own test suite and the TCK.

Thoughts?

**Comment by mnriem** [comment]
@Thihup Feel free to create a 'spring' branch for this experimentation. If you can get it to work then once it is done we can figure out where it should land. That OK?

**Reply by Thihup** [reply]
Do I need to create an issue for each commit in this branch too?

**Reply by mnriem** [reply]
No that won't be necessary as you are prototyping so really once it is all done one issue would be sufficient and we will all look it over and be amazed! ;)
